Add dialogs and refactor modules for path generation

inchkev opened the pull request:
* Refactored Chris's path generation code, rewrote many parts
* Created `mathutils.py` and `pathutils.py` for vector math and path generation code, respectively
* Added `pathgen_toolbar.py`, which has individual dialogs to generate actions (paths) for cylinder, helix, sphere, line, and single point
  * Used `FancyTextCtrl` (also used in properties panel) for auto unit conversion
  * Got icons for the toolbar, created custom icon for helix path
  * Should for the most part resolve #56
* Viewport now defaults to not showing any actions (`update_test()` in `core.py` is finally removed 🎉)
* Renamed `toolbar.py` which had baud/port controls to `serial_toolbar.py`

djihbril commented:
2 things:

> Viewport now defaults to not showing any actions (`update_test()` in `core.py` can finally be deleted 🎉)

I have deleted that method and replaced it with one that plots the cameras from `machine.ini` onto the viewport. Do we not want to plot the cameras on start up? if not when should we do it?
> Renamed `toolbar.py` which had baud/port controls to `serial_toolbar.py`

Does that toolbar only handle serial?

inchkev commented:
> I have deleted that method and replaced it with one that plots the cameras from machine.ini onto the viewport. Do we not want to plot the cameras on start up? if not when should we do it?

My bad, could have explained that better - there's no more `update_test()` to manually create a path, but now there's `update_devices()` which still initializes the cameras. So cameras *should* still be plotted on startup.

> Does that toolbar only handle serial?

You're right, it also has start/stop/pause to the machine (not just serial). Have any suggestions for a new name? Maybe `machine_toolbar` or `connect_toolbar`?

inchkev commented:
> I think `machine_toolbar` is fine for now.

Alright, renamed!

djihbril commented on README.md:
````diff
@@ -26,13 +26,13 @@ __Note: This project supports Python version 3.x.x.__
 ```
-__Note: These 2 commands can be run from any folder in the project directory tree__
+*Note: These 2 commands can be run from any folder in the project directory tree.*
 ### Configuration File
 
 The configuration file `config.ini` contains initialization settings.
````
```suggestion
### Configuration Files

* `config.ini` contains initialization settings.
* `machine.ini` contains settings for the imaging instrument.
```

inchkev replied:
Looks good, just committed those changes.
